### DISCUSSION TITLE
crypto_kx_server_session_keys & crypto_kx_client_session_keys

### DISCUSSION THREAD

**Discussion by shmuelsilverman** [jedisct1/libsodium#1185] [Q&A]
While developing a higher-level API I thought that these two methods/functions are the same code with symmetric parameters. Assuming this is correct, wouldn't it be more correct for one to simply invoke the other and have a single function?
I ask this to make sure I understand it correctly and not make a mistake in my API.
This code change will not impact existing applications and will not change the library API.
Kind thanks.

**Comment by jedisct1** [comment]
Hi!

This is correct, both functions return the same set of keys.

There are distinct functions instead of a unique one for clarity.

For a given function, the parameters are always the same, always in the same order.

So once one of the party has been chosen as being the server, and the other one the client, there is no confusion as to how to call the function, and which key in the output to use in each direction.

With a single function, one needs to think about when to reverse the parameters or not, and the output keys don't have an explicit label.
